### PR TITLE
Usage examples 1 - Kartik Kaushik

### PR DESCRIPTION
**Pull request by kay-kaushik**
# Description 

This pull request features usage exaples for create_sprite(), draw_sprite(), move_sprite(), free_sprite() and key_typed() functions. They are really detailed examples with descriptive gifs and images. The usage examples are in cpp language.

# changes

I have created 3 files per function including txt, png and cpp files.

#  testing 

I tested all the changes made to the code on my machine before creating the pull request.

#  benefits

These high quality usage examples give more context to the functions. This will help users reading the documentation to better understand the function. This will result in faster development.

### CODE REVIEW

**Comment by kay-kaushik** [comment]
[Create Code Usage Examples for:](https://trello.com/c/AaJdb3Pr/596-create-code-usage-examples-for)

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Usage Example: creating a sprite
+ **/
+
+#include "splashkit.h"
+
+int main()
+{
+    // widow to draw the sprite on
```
Comment (hugglesfox) [inline src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.cpp:9]:
```suggestion
    // window to draw the sprite on
```

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    window start = open_window("create_sprite", 600, 600);
+
+    // bitmap for creating a sprite
+    bitmap player = load_bitmap("playerBmp", "protSpriteSheet.png");
+    bitmap_set_cell_details(player, 31, 32, 4, 3, 12);
+
+    // creating the player sprite
+    sprite player_sprite = create_sprite(player);
+
+    // setting the co-ordinates in refrence to the window
```
Comment (hugglesfox) [inline src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.cpp:19]: Isn't this centering the sprite on the window, rather then setting it's coordinates relative to?

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+### key_typed() usage exaple
```
Comment (hugglesfox) [inline src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt:1]:
```suggestion
### key_typed() usage example
```

This mistake has been made in the other example descriptions too

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt
Comment (hugglesfox) [inline src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt:1]: Should this be an mdx file rather then a txt?

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+The following code is an example of using [key typed](https://splashkit.io/api/input/#key-typed) to move a sprite in a game window to the right after right key is typed.
```
Comment (hugglesfox) [inline src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt:3]:
```suggestion
The following code is an example of using [`key_typed`](https://splashkit.io/api/input/#key-typed) to move a sprite in a game window to the right after right key is typed.
```

Should probably run this past @omckeon, however my preference would be to have the function name in snake_case like how it is in the code and to wrap it in a code block. Same could be done for the page title.

File: src/assets/usage-examples-code/sprites/draw_sprite/draw_sprite-1-drawing-sprite.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    sprite player_sprite = create_sprite(player);
+
+    // setting the co-ordinates in refrence to the window
+    sprite_set_x(player_sprite, 300);
+    sprite_set_y(player_sprite, 300);
+    
+    // clearing the screen
+    clear_screen(COLOR_BLACK);
+
+    //drawing the sprite
```
Comment (hugglesfox) [inline src/assets/usage-examples-code/sprites/draw_sprite/draw_sprite-1-drawing-sprite.cpp:26]:
```suggestion
    // drawing the sprite
```

**Review by hugglesfox — CHANGES_REQUESTED** [review]
Looks good, just a couple of questions and spelling mistakes. 

However it looks like you've created your usage-examples branch from the branch which you fixed the dead links on, meaning that those commits have been included in the PR. It would be nice to remove the commits fixing the dead links from this PR (hint: use `git rebase -i`).

**Comment by lexlam1524** [comment]
[Naming refactor of ExecutionEnvironment and related code](https://trello.com/c/mqHADqTC/584-naming-refactor-of-executionenvironment-and-related-code)

File: src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt
Comment (omckeon) [reply src/assets/usage-examples-code/input/key_typed/key_typed-1-type-key.txt:1]: I've requested these files to be .txt for now. They would then be used in a script to generate the .mdx files.

**Comment by omckeon** [comment]
Just closing PRs that can't/won't be merged (due to formatting or other issues), this is separate from any capstone grading.

**Pull request closed without merging**